tools/HogEdit/HogEdit.cpp

```diff
@@ -21,102 +21,278 @@
-              << std::endl;
-    return 1;
-  }
+template<std::size_t len>
+std::string arr_to_str(const std::array<char, len>& arr) {
+  return std::string(std::data(arr), strnlen(std::data(arr), std::size(arr)) );
+}
 
-  std::filesystem::path hog_filename{argv[1]};
-  std::filesystem::path input_filename{argv[2]};
-  if (!is_regular_file(input_filename)) {
-    std::cout << input_filename << " is not a regular file!" << std::endl;
-    return 1;
-  }
+D3::HogFormat::hog_iter find_file_entry(D3::HogFormat& format, const std::string& name)
+{
+  auto fn = [&name](const auto& hfe) { return lowercase(name) == lowercase(arr_to_str(hfe.name)); };
+  if (auto it = std::find_if(std::begin(format), std::end(format), fn); it != std::end(format))
+    return it;
+  return std::end(format);
+}
 
-  try {
-    create_directories(absolute(hog_filename).parent_path());
-    std::ifstream input(input_filename, std::ios_base::in);
-    auto parent_input_path = input_filename.parent_path();
-    for (std::string line; std::getline(input, line);) {
-      if (!is_regular_file(parent_input_path / line)) {
-        std::cout << "Warning! File " << line << " from " << input_filename << " does not exists! Skipping..."
-                  << std::endl;
-      } else if (std::filesystem::path(line).filename().string().size() > 36) {
-        std::cout << "Warning! Length of name of file " << line << " is more than 36 symbols! Skipping..." << std::endl;
-      } else {
-        input_files.emplace_back(parent_input_path / line);
-      }
+std::list<std::filesystem::path> validate_files(const std::filesystem::path& base_path, std::list<std::filesystem::path> filepaths)
+{
+  std::list<std::filesystem::path> valid_files;
+  for(auto& filepath : filepaths) {
+    if (!std::filesystem::exists(base_path / filepath)) {
+      std::cout << "File " << (base_path / filepath) << " does not exist! Skipping... " << std::endl;
     }
+    else {
+      if (!std::filesystem::is_regular_file(base_path / filepath))
+       throw std::runtime_error(filepath.string() + " is not a regular file.");
 
-  } catch (std::exception &e) {
-    std::cout << "Exception: " << e.what() << std::endl;
-    return 1;
+      if (filepath.string().size() > 36)
+        throw std::runtime_error("Length of filepath \"" + filepath.string() + "\" exceeds 36 bytes.");
+
+      valid_files.emplace_back(filepath);
+    }
   }
+  return valid_files;
+}
+
 
-  auto customLess = [](const std::filesystem::path &a, const std::filesystem::path &b) {
-    return (str_tolower(a.u8string()) < str_tolower(b.u8string()));
-  };
+namespace hog
+{
+  std::filesystem::path archive_path; // path of hog archive
+  std::filesystem::path base_path = std::filesystem::current_path();
+  std::list<std::filesystem::path> file_path_list;
 
-  std::sort(input_files.begin(), input_files.end(), customLess);
 
-  D3::HogFormat hog_table;
+  void sort_file_list(void) {
```

Comment:
Redundant void argument list in function definition.

Reply:
It makes it easy to tell definitions apart from invocations.